AspNetCore3ChildlifetimeScope/Program.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Threading.Tasks;
+using AspNetCore3ChildlifetimeScope.Dependencies;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace AspNetCore3ChildlifetimeScope
+{
+    public static class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var containerBuilder = new ContainerBuilder();
+
+            containerBuilder.RegisterType<SingletonRootDependency>()
+                .As<ISingletonRootDependency>()
+                .AsSelf()
+                .SingleInstance();
+
+            var container = containerBuilder.Build();
+
+            var scope = container.BeginLifetimeScope("RootChildScope", builder => builder.RegisterType<TransientRootChildScopeDependency>().As<ITransientRootChildScopeDependency>().AsSelf().InstancePerDependency());
+
+            using var hostApplicationA = CreateHostForApplicationA(scope, args);
+
+            using var hostApplicationB = CreateHostForApplicationB(scope, args);
+
+            await Task.WhenAll(hostApplicationA.RunAsync(), hostApplicationB.RunAsync());
+        }
+
+        private static IHost CreateHostForApplicationA(ILifetimeScope container, string[] args)
```

Comment:
It might be better to have these `CreateHost` calls be methods with actual curly braces.

Doing that allows you to more easily put inline comments, which is definitely needed here - this is pretty key for the demo.

```c#
// This shows how each application can use a separate lifetime scope
// as the "root" for the application, whilst sharing a common container
// for common dependencies. Each application can get app-specific
// dependencies, but common items can be registered outside the app.
```

Reply:
Unclear if that's the right line-wrapping or whatever.